### PR TITLE
feat(adapters): reconcile MockProductConfig schema (Phase 2 of #996)

### PR DESCRIPTION
**Pull request by bokelley**
## Summary

Picks up the dormant Phase 2 work from issue #996 ("Adapter Schema System") for the Mock adapter. First of three sub-issues under #1239 — Broadstreet and GAM follow in their own PRs once the pattern is approved here.

Phase 1 (#1007) registered `MockProductConfig` as `product_config_class` but the schema was 5 fields with fractional units (0–1) while the admin form actually saved a 13-field percent-units (0–100) shape that nothing validated against. The legacy `validate_product_config` method also had a tuple-truthiness bug (returns `tuple[bool, str | None]`, call site treated it as a list of error strings) — meaning validation **never fired**.

This PR collapses the two-source-of-truth into a single `MockProductConfig` used at both the form save boundary AND the runtime adapter contract.

## What changed

**`src/adapters/mock_ad_server.py`**
- `MockProductConfig`: 13 fields with percent units, `adapter_type: Literal["mock"]` discriminator, nested `MockDeliverySimulation` (with `extra="forbid"`), `test_mode: Literal[...]` (replaces hand-written `field_validator`), real Pydantic range validators on every numeric
- `parse_implementation_config()`: typed parser, returns default-constructed instance on empty/None, validates strict otherwise
- Admin POST handler: replaces inline dict-build + broken legacy validator with `try`/`except ValidationError`. **Submitted format selections now survive validation failures** (used to silently fall back to DB-stored selection). Error messages include field path via `loc` prefix.
- `validate_product_config()`: delegates to `parse_implementation_config` so the helper is wired in `src/`. Preserves base-class `tuple[bool, str | None]` contract until Broadstreet+GAM reconciliation completes.

**`tests/unit/adapters/test_schemas.py` and `tests/unit/admin/test_adapter_blueprints.py`**
- Fraction-based assertions → percent-based shape
- Parametrized range validation across 8 boundary cases
- Nested `MockDeliverySimulation` `extra="forbid"` coverage
- Round-trip test extended to validate JSONType-shaped dict→model→dict for nested `delivery_simulation`
- New `TestParseMockImplementationConfig`: helper coverage (None default, partial overlay, ValidationError propagation, round-trip)

## Reviewer feedback applied (Round 1)

`code-reviewer` and `python-expert` both reviewed before push. All block-merge and should-fix items applied:
- Format selection bug on validation failure → fixed (read submitted formats before try, pass to both branches)
- ValidationError messages losing field context → fixed (`loc` prefix in join)
- Outer-strict / inner-permissive footgun on `MockDeliverySimulation` → fixed (explicit `ConfigDict(extra="forbid")`)
- Dead `parse_implementation_config` in `src/` → fixed (wired through `validate_product_config` override)
- Test data sloppiness: `error_rate=200` against `le=100` → fixed (`100.1`)
- `test_mode: str + field_validator` → `test_mode: Literal[...]` (cleaner JSON schema, removes hand-written validator)

## Test plan

- [x] `make quality` clean (4298 unit tests pass, ruff format, ruff check, mypy)
- [x] `tox -e admin` ✓
- [x] `tox -e ui` ✓
- [x] `tox -e e2e` ✓
- [x] `tox -e unit` ✓
- [x] Targeted: `tests/integration/test_list_creatives_auth.py` and the broader Mock surface pass in isolation
- [ ] CI re-runs the full integration suite

**Note on `tox -e integration`:** the full integration suite has pre-existing test pollution that surfaces flaky failures in different tests on each run, irrespective of this PR's diff. Verified by running tox integration on HEAD~1 (without my changes): 1 setup error in `test_creative_formats_id_filters.py`. At HEAD with my changes: 3 failures in `test_list_creatives_auth.py`. Same total test count both runs; tests that fail in the suite all pass in isolation; Mock schema diff has zero overlap with the failing tests' code paths. Pushed with `--no-verify` after authorization. Worth filing a separate tracking issue for the integration pollution.

## Out of scope (deferred)

- **Broadstreet reconciliation** (`BroadstreetImplementationConfig` → `BroadstreetProductConfig` at the registered slot) — separate sub-issue under #1239
- **GAM reconciliation** — schema needs `supported_format_types` and `time_zone` added (surfaced during the audit) plus 30+ adapter-boundary read-site rewrites
- **Migration-as-audit** — Alembic step that validates all existing `Product.implementation_config` rows against the reconciled schemas
- **Phase 3 of #996**: dynamic admin form generation from `model_json_schema()`

## Refs

- Parent: #996 (reopened — was incorrectly closed during a triage sweep)
- Sub-issue tracker: #1239
- Phase 1: #1007 (merged)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/adapters/mock_ad_server.py [hunk 1]
```diff
@@ -40,14 +40,117 @@ class MockConnectionConfig(BaseConnectionConfig):
...
+    )
+    error_rate: float = Field(
+        default=0.1,
+        ge=0.0,
+        le=100.0,
+        description="Simulated error rate as percent (0-100)",
+    )
+
+    # Behavior modes
+    test_mode: Literal["normal", "slow", "fast", "uneven", "error"] = Field(
```
Comment (ChrisHuie) [inline src/adapters/mock_ad_server.py:114]: This vocabulary doesn't match except for "normal" and think will fail deploy. 

Did you mean `test_mode: Literal["normal", "high_demand", "degraded", "outage"]` ?

**Review by KonstantinMirin — COMMENTED** [review]
See architectural feedback on #1239 for the broader context.

**`test_mode` values diverge from the form.** The schema defines `Literal["normal", "slow", "fast", "uneven", "error"]` but the admin template and blueprint use `["normal", "high_demand", "degraded", "outage"]`. Three of four selectable options in the UI will raise `ValidationError`. This is what happens when validation is added at the read site without constraining the write path — the form and schema were reconciled independently and they disagree.

**Duplicate route registration.** Both `adapters_bp` and `register_ui_routes` register the same URL. The blueprint handler is dead code with different validation behavior. Should be removed.

**Fractional→percentage silent breakage.** Stored `fill_rate=0.85` (from Phase 1) passes `ge=0.0, le=100.0` but is now interpreted as 0.85% instead of 85%. Needs a data migration before this schema is enforced — not after.

**Comment by bokelley** [comment]
Thanks for the substantive review @KonstantinMirin — the architectural critique on #1239 (validate-at-write per Pattern #7, single-source-of-truth for validators, data migration before schema tightening) plus the specific bugs flagged here are all well-grounded.

The rework path you outlined is best driven by maintainers who know the convention history. Closing this PR rather than pushing back from outside.

Branches and the feedback docs in `.context/` stay in the repo as reference if any of the schema design decisions or the upstream-SDK feedback (`adcp-v4-migration-feedback.md`, `adcp-decisioning-design-feedback.md`) are useful for future work.
